app/WebParsing/ReqParser.hs

```diff
@@ -402,10 +402,22 @@ programOrParser = do
         [x] -> return x
         (x:xs) -> return $ OR $ flattenOr (x:xs)
 
+-- | Parser for FCE requirements following the dept name
+-- | eg. "... 9.0 CSC FCEs ..."
+deptBefFcesParser :: Parser Req
```

Comment:
You don't need to create a separate parsing function here. Use an [`optionMaybe`](https://hackage.haskell.org/package/parsec-3.1.15.1/docs/Text-Parsec.html#v:optionMaybe) in the `fcesParser` to attempt to parse a department after the credit number, and then you can use an `if` expression to branch on whether that parsing succeeded or failed.